src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -100,25 +100,27 @@ public Config validate() {
       return new Config(validations);
     }
 
-    try (RestHighLevelClient client = clientFactory.client()) {
-      validateCredentials();
-      validateDataStreamConfigs();
-      validateIgnoreConfigs();
-      validateKerberos();
-      validateLingerMs();
-      validateMaxBufferedRecords();
-      validateProxy();
-      validateSsl();
-
-      if (!hasErrors()) {
-        // no point if previous configs are invalid
+    validateCredentials();
+    validateDataStreamConfigs();
+    validateIgnoreConfigs();
+    validateKerberos();
+    validateLingerMs();
+    validateMaxBufferedRecords();
+    validateProxy();
+    validateSsl();
+
+    if (!hasErrors()) {
+      // no point in connection validation if previous ones fails
+      try (RestHighLevelClient client = clientFactory.client()) {
         validateConnection(client);
-      }
-      if (!hasErrors()) {
         validateVersion(client);
+      } catch (IOException e) {
+        log.warn("Closing the client failed.", e);
```

Comment:
nit: I know this was here before, but seems kind of weird that it assumes that closinbg the client was the problem, doesn't it?

Reply:
This is because the other exceptions in the client methods are captured in their respective validations. So only close can throw an IOException here.